level2.src.html

```diff
@@ -84,29 +84,90 @@ <h2 id="algorithms">Algorithms</h2>
     <h3 id="upgrade-algorithm">Upgrade |request| to an <i lang="la">a priori</i>
     authenticated URL as mixed content, if appropriate</h3>
 
-    TODO. This algorithm is called from Fetch before blocking requests as mixed content. It checks if the request is optionally-blockable mixed content, and if so rewrites the scheme to HTTPS.
+    Note: The Fetch specification will hook into this algorithm to upgrade optionally-blockable
+    mixed content to HTTPS.
+
+    Given a <a>Request</a> <var>request</var>, this algorithm will rewrite
+    its <a for="request">url</a> if the request is deemed to be optionally-blockable mixed content,
+    via the following algorithm:
+
+    <ol>
+      <li>
+        If one or more of the following conditions is met, return without modifying <var>request</var>:
+        <ol>
+          <li>
+            [[mixed-content#categorize-settings-object]] returns "<code>Does Not Restrict Mixed Security
+            Contents</code>" when applied to <var>request</var>'s <a for="request">client</a>.
+          </li>
+          <li>
+            <var>request</var>'s <a for="request">mode</a> is <code>CORS</code>
+            or <code>CORS-with-forced-preflight</code>.
+          </li>
+          <li>
+            <var>request</var>'s <a for="request">destination</a> is not "<code>image</code>",
+            "<code>audio</code>", or "<code>video</code>".
+          </li>
+          <li>
+            <var>request</var>'s <a for="request">destination</a> is "<code>image</code>"
+            and <var>request</var>'s <a for="request">initiator</a> is "<code>imageset</code>".
+          </li>
+          <li>
+            <var>request</var>'s <a for="request">url</a> is an
+            [[mixed-content#a-priori-authenticated-url]].
```

Comment:
Nit: Maybe move this one up, since it's probably the most common case these days?

Reply:
Done